Samples/Flix/src/main/java/dev/testify/samples/flix/domain/model/mapper/DataLayerEntityToDomainModelMapper.kt

```diff
@@ -0,0 +1,162 @@
+package dev.testify.samples.flix.domain.model.mapper
+
+import dev.testify.samples.flix.data.remote.tmdb.TheMovieDbUrlResolver
+import dev.testify.samples.flix.data.remote.tmdb.entity.*
+import dev.testify.samples.flix.domain.model.*
+import kotlinx.datetime.Instant
+
+interface DataLayerEntityToDomainModelMapper {
+    fun map(movie: Movie, urlResolver: TheMovieDbUrlResolver): MovieDomainModel?
+    fun map(movieDetails: MovieDetail, urlResolver: TheMovieDbUrlResolver): MovieDomainModel?
+    fun map(movieReleaseDates: MovieReleaseDates): MovieReleaseDatesDomainModel?
+    fun map(movieCredits: MovieCredits, urlResolver: TheMovieDbUrlResolver): MovieCreditsDomainModel
+}
+
+class DataLayerEntityToDomainModelMapperImpl : DataLayerEntityToDomainModelMapper {
```

Comment:
Also not a huge fan of `Impl` classes... would rather `DataLayerEntityToDomainModelMapperInterface` and `DataLayerEntityToDomainModelMapper` but I suppose it depends which one you reference elsewhere.

Reply:
I understand. It's coming from me setting up dependency injection. I could just drop the interfaces. I just wanted to play with the injection.